Python/Product/PythonTools/PythonTools/Debugger/DebugAdapterLauncher.cs

```diff
@@ -50,24 +52,45 @@ public sealed class DebugAdapterLauncher : IAdapterLauncher {
         public void UpdateLaunchOptions(IAdapterLaunchInfo launchInfo) {
             if (launchInfo.LaunchType == LaunchType.Attach) {
                 JObject launchJson = new JObject();
-
                 launchInfo.DebugPort.GetPortName(out string uri);
+
                 launchJson["remote"] = uri;
+                AddDebugOptions(launchJson);
+                AddRules(launchJson);
 
-                var debugService = (IPythonDebugOptionsService)Package.GetGlobalService(typeof(IPythonDebugOptionsService));
-                JArray debugOptions = new JArray();
-                if (debugService.ShowFunctionReturnValue) {
-                    debugOptions.Add("ShowReturnValue");
-                }
-                if (debugService.BreakOnSystemExitZero) {
-                    debugOptions.Add("BreakOnSystemExitZero");
-                }
-
-                if (debugOptions.Count > 0) {
-                    launchJson["debugOptions"] = debugOptions;
-                }
                 launchInfo.LaunchJson = launchJson.ToString();
             }
         }
+
+        private void AddDebugOptions(JObject launchJson) {
+            var debugService = (IPythonDebugOptionsService)Package.GetGlobalService(typeof(IPythonDebugOptionsService));
+
+            JArray debugOptions = new JArray();
+            if (debugService.ShowFunctionReturnValue) {
+                debugOptions.Add("ShowReturnValue");
+            }
+            if (debugService.BreakOnSystemExitZero) {
+                debugOptions.Add("BreakOnSystemExitZero");
+            }
+
+            if (debugOptions.Count > 0) {
+                launchJson["debugOptions"] = debugOptions;
+            }
+        }
+
+        [Conditional("DEBUG")]
```

Comment:
This attribute isn't right. We want this to work in release builds of PTVS as well.